### PR TITLE
Add helpers for authenticated media, and associated documentation

### PR DESCRIPTION
**Pull request by turt2live**
## Checklist

-   [x] Tests written for new code (and old code if feasible).
-   [x] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,6 +21,10 @@ endpoints from before Matrix 1.1, for example.
 
 # Quickstart
 
+> [!IMPORTANT]
```
Comment (turt2live) [inline README.md:24]: for the reviewer - I think this our first usage of the blockquote styles provided by github. See https://github.com/orgs/community/discussions/16925 for details.

**Review by dbkr — COMMENTED** [review]
How will we manage the second transition from unstable to stable?

**Comment by turt2live** [comment]
Ah, sorry, it's mentioned briefly in the README diff here and in the depths of https://github.com/element-hq/element-web/pull/27326#discussion_r1576303882 somewhere - we'd be introducing a breaking change later (once the MSC is either stable or released) which *only* returns authenticated media URLs. 

I'm not sure there's much of a way to avoid a breaking change here, given we'd want the js-sdk to produce authentication-required URLs by default as soon as it possibly can.

**Comment by dbkr** [comment]
Fair enough, as long as there's a plan for managing the second transition (ie. to stable) as well.

**Pull request merged**